flink-core/src/test/java/org/apache/flink/api/java/typeutils/LambdaExtractionTest.java

```diff
@@ -50,12 +41,12 @@
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertNull;
-import static org.junit.Assert.fail;
+import static org.junit.Assert.assertTrue;
 
 /**
- * Tests the type extractor for lambda functions.
+ * Tests the type extractor for lambda functions. Many tests only work if the compiler supports
```

Comment:
Some tests, like `flatMapLambda()` are not there anymore? It's because we don't use Tycho anymore but do theses tests now reside somewhere else? If not, we should probably have a new module `flink-tycho-lambda-tests` where we test those things that only work with Tycho. Otherwise we lose that coverage.

Reply:
A flat map can not work by definition because it contains generics. I will add some tests for the `returns`.